### PR TITLE
SPAKE2P Parameters Tool.

### PR DESCRIPTION
**Pull request by emargolis**
#### Problem
Need a tool to generate Setup PIN Code and Verifier. This might be needed for testing purposes as well as for the production parameters generation for device manufacture provisioning. 

#### Change overview
Added command to calculate Spake2p Verifier from Setup Code, Iteration Counter, and the Salt.

Refer to README.md for more details.

#### Testing

### CODE REVIEW

File: src/tools/spake2p/Cmd_GenVerifier.cpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    "          * 12345678\n"
+    "          * 87654321\n"
+    "\n"
+    "   -i, --iteration-count <int>\n"
+    "\n"
+    "       SPAKE2P PBKDF iteration count. The value should be positive integer in range [1000..100000].\n"
+    "\n"
+    "   -s, --salt <string>\n"
+    "\n"
+    "       SPAKE2P PBKDF salt input value. Legth of salt string should be in range [16..32].\n"
```
Comment (balducci-apple) [inline src/tools/spake2p/Cmd_GenVerifier.cpp:90]:
```suggestion
    "       SPAKE2P PBKDF salt input value. Length of salt string should be in range [16..32].\n"
```

File: src/tools/spake2p/Cmd_GenVerifier.cpp [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    "\n"
+    "   -o, --out <file>\n"
+    "\n"
+    "       File to contain the new certificate. Specify '-' for stdout.\n"
```
Comment (balducci-apple) [inline src/tools/spake2p/Cmd_GenVerifier.cpp:94]: I don't think certificate is the right output. We should also explicitly state the output format of the base64 encoded verifier.

File: src/tools/spake2p/Cmd_GenVerifier.cpp (hunk 1 above)
Comment (balducci-apple) [inline src/tools/spake2p/Cmd_GenVerifier.cpp:90]: The salt should be unique per PIN code generated. Should we clarify that these will be randomized (after the first) if count > 1 ?

File: src/tools/spake2p/Cmd_GenVerifier.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    if (fprintf(outFile, "Index,Iteration Count, Salt, PIN Code, Verifier\n") < 0 || ferror(outFile))
+    {
+        fprintf(stderr, "Error writing to output file: %s\n", strerror(errno));
+        return false;
+    }
+
+    for (uint32_t i = 0; i < gCount; i++)
+    {
+        chip::PASEVerifier verifier;
+        CHIP_ERROR err = chip::PASESession::GeneratePASEVerifier(
```
Comment (balducci-apple) [inline src/tools/spake2p/Cmd_GenVerifier.cpp:234]: We should generate random salts after the first.

**Review by balducci-apple — DISMISSED** [review]

File: src/tools/spake2p/Cmd_GenVerifier.cpp (hunk 3 above)
Comment (emargolis) [reply src/tools/spake2p/Cmd_GenVerifier.cpp:234]: done. Added randomization option for salt parameters

File: src/tools/spake2p/Cmd_GenVerifier.cpp (hunk 1 above)
Comment (emargolis) [reply src/tools/spake2p/Cmd_GenVerifier.cpp:90]: done

File: src/tools/spake2p/Cmd_GenVerifier.cpp (hunk 2 above)
Comment (emargolis) [reply src/tools/spake2p/Cmd_GenVerifier.cpp:94]: doen

File: src/tools/spake2p/Cmd_GenVerifier.cpp (hunk 1 above)
Comment (emargolis) [reply src/tools/spake2p/Cmd_GenVerifier.cpp:90]: done

**Review by woody-apple — APPROVED** [review]
Fast tracking, given adding tooling.

**Pull request merged**